### ISSUE TITLE
Keyword with spaces

### ISSUE THREAD

**Issue by jiegen78** [Hootrix/keyword_alert_bot#12]
is it possible to keyword a whole sentence?
or, is it possible to make a filter "word 1 but not word 2" i.e. send notification if there is only word 1 but not word 2

thanks

**Comment by Hootrix** [comment]
1. the entire sentence can be used as a keyword 
2. Use regular expressions for exclusion

Advanced features require a little knowledge of regex basics. The keyword exclusion function can be arranged at leisure 😁

**Comment by jiegen78** [comment]
i am not a regex expert but i digged the internet for a few hours for a regex "with js syntax" and found nothing working.
can you kindly make me an example of the syntax?
if I put /this is a sentence/ig does not work. 
thanks a lot

**Comment by Hootrix** [comment]
> i am not a regex expert but i digged the internet for a few hours for a regex "with js syntax" and found nothing working. can you kindly make me an example of the syntax? if I put /this is a sentence/ig does not work. thanks a lot

try it.
/^\s*this is a sentence\s*$/ig

**Comment by jiegen78** [comment]
thanks for the example, it works on a regex test site,
but if I put in the bot the command /subscribe /\s*sentence with spaces\s*/ig the command is not accepted (also with ^ and $ ) 
it looks like the bot uses space as a separator and does not read correctly the command after the first space, resulting into an incomplete command.

**Comment by Hootrix** [comment]
> thanks for the example, it works on a regex test site,
> 
> but if I put in the bot the command /subscribe /\s*sentence with spaces\s*/ig the command is not accepted (also with ^ and $ ) 
> 
> it looks like the bot uses space as a separator and does not read correctly the command after the first space, resulting into an incomplete command.

yep. Spaces are indeed separators. I need to see the full command you entered. It may need to be fixed

**Comment by jiegen78** [comment]
you can consider "/subscribe /\s*sentence with spaces\s*/i testchannel"  as a non-specific copy of my command, 
I have not used ^ and $ because the sentence with spaces is part of the whole message I have to filter.

thanks a lot.

**Comment by Hootrix** [comment]
Sorry, the use of spaces does have a bug.

You can temporarily replace spaces with \s

example：

/subscribe /\bsentencs\s+with\s+spaces\b/i  test-channel

**Comment by jiegen78** [comment]
this works,
thanks a lot
